### ISSUE TITLE
Mpeg streaming crash with Apache auth - 1.28

### ISSUE THREAD

**Issue by sbmechanics** [ZoneMinder/ZoneMinder#585]
Mpeg streaming using Apache LDAP auth was working with no difficulty until 1.28.  Nph-zms now crashes when this external auth is used.  Using no auth or internal auth allows streaming without difficulty.  I have made no changes to Apache or ZM configs since an upgrade from 1.27.

2014-11-18 15:29:10.929333	web_js	29954	ERR	getStreamCmdResponse stream error: socket_sendto( /tmp/zm/zms-690136s.sock ) failed: Connection refused - checkStreamForErrors()	?view=watch	
2014-11-18 15:29:10.653644	web_php	29953	ERR	socket_sendto( /tmp/zm/zms-690136s.sock ) failed: Connection refused	includes/functions.php	2330
2014-11-18 15:29:07.894027	zms	30054	INF	addr2line -e /var/www/cgi-bin/nph-zms 0x47b56e 0x3cf420f710 0x3cf3a897cb 0x4587f8 0x451e01 0x452f93 0x407b7d 0x3cf3a1ed5d 0x406b39	zm_signal.cpp	106
2014-11-18 15:29:07.891262	zms	30054	INF	Backtrace complete, please execute the following command for more information	zm_signal.cpp	105
2014-11-18 15:29:07.888837	zms	30054	ERR	Backtrace 8: /var/www/cgi-bin/nph-zms() [0x406b39]	zm_signal.cpp	98
2014-11-18 15:29:07.886223	zms	30054	ERR	Backtrace 7: /lib64/libc.so.6(__libc_start_main+0xfd) [0x3cf3a1ed5d]	zm_signal.cpp	98
2014-11-18 15:29:07.883494	zms	30054	ERR	Backtrace 6: /var/www/cgi-bin/nph-zms() [0x407b7d]	zm_signal.cpp	98
2014-11-18 15:29:07.880901	zms	30054	ERR	Backtrace 5: /var/www/cgi-bin/nph-zms() [0x452f93]	zm_signal.cpp	98
2014-11-18 15:29:07.878262	zms	30054	ERR	Backtrace 4: /var/www/cgi-bin/nph-zms() [0x451e01]	zm_signal.cpp	98
2014-11-18 15:29:07.875812	zms	30054	ERR	Backtrace 3: /var/www/cgi-bin/nph-zms() [0x4587f8]	zm_signal.cpp	98
2014-11-18 15:29:07.872187	zms	30054	ERR	Backtrace 2: /lib64/libc.so.6(memcpy+0x15b) [0x3cf3a897cb]	zm_signal.cpp	98
2014-11-18 15:29:07.870048	zms	30054	ERR	Backtrace 1: /lib64/libpthread.so.0() [0x3cf420f710]	zm_signal.cpp	98
2014-11-18 15:29:07.866766	zms	30054	ERR	Backtrace 0: /var/www/cgi-bin/nph-zms() [0x47b56e]	zm_signal.cpp	98
2014-11-18 15:29:07.862593	zms	30054	ERR	Signal address is (nil), from 0x3cf3a897cb	zm_signal.cpp	77
2014-11-18 15:29:07.852237	zms	30054	ERR	Got signal 11 (Segmentation fault), crashing	zm_signal.cpp	54
2014-11-18 15:29:07.777370	zms	30054	WAR	Shared data not initialised by capture daemon, some query functions may not be available or produce invalid results	zm_monitor.cpp	424
2014-11-18 15:29:07.702241	zms	30054	INF	Authenticated user 'ldaptestuser'	zm_user.cpp	140
2014-11-18 15:29:07.245605	web_js	29956	WAR	Adjusting to minimum width when getting popup size for tag 'watch' - getPopupSize()		
2014-11-18 15:29:05.022084	zmc_m1	30029	INF	Starting Capture	zmc.cpp	195
2014-11-18 15:29:04.927440	zmdc	29871	WAR	Can't find process with command of 'zma -m 1'	zmdc.pl	
2014-11-18 15:29:04.836140	zmdc	30029	INF	'zmc -m 1' started at 14/11/18 15:29:04	zmdc.pl	
2014-11-18 15:29:04.834100	zmdc	29871	INF	'zmc -m 1' starting at 14/11/18 15:29:04, pid = 30029	zmdc.pl	
2014-11-18 15:29:04.829980	zmdc	29871	INF	Starting pending process, zmc -m 1	zmdc.pl	
2014-11-18 15:29:04.827730	zmdc	29871	ERR	'zmc -m 1' exited abnormally, exit status 255	zmdc.pl	
2014-11-18 15:29:04.801750	zmdc	29871	WAR	Can't find process with command of 'zmtrack.pl -m 1'	zmdc.pl	
2014-11-18 15:29:04.652540	zmdc	30015	INF	'zmc -m 1' started at 14/11/18 15:29:04	zmdc.pl	
2014-11-18 15:29:04.652510	zmdc	29871	INF	'zmc -m 1' starting at 14/11/18 15:29:04, pid = 30015	zmdc.pl	
2014-11-18 15:29:04.525210	zmdc	29871	WAR	Can't find process with command of 'zmc -m 1'	zmdc.pl	
2014-11-18 15:29:04.401500	zmdc	29871	WAR	Can't find process with command of 'zma -m 1'	zmdc.pl	
2014-11-18 15:29:04.278040	zmdc	29871	WAR	Can't find process with command of 'zmtrack.pl -m 1'	zmdc.pl	
2014-11-18 15:28:38.555890	zmwatch	29928	INF	Watchdog pausing for 30 seconds	zmwatch.pl	
2014-11-18 15:28:38.545590	zmwatch	29928	INF	Watchdog starting

System config:
Linux 2.6.32-504.1.3.el6.x86_64 #1 SMP Tue Nov 11 17:57:25 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
CentOS release 6.6 (Final)
Server version: Apache/2.2.15 (Unix)
Server built:   Oct 16 2014 14:48:21

Auth info:
AUTH_TYPE	remote
AUTH_RELAY	none

Configure string:
./configure --with-webdir=/var/www/html/zm --with-cgidir=/var/www/cgi-bin --with-webuser=apache --with-webgroup=apache ZM_DB_HOST=localhost ZM_DB_NAME=zm ZM_DB_USER=zmuser ZM_DB_PASS=zmpass ZM_SSL_LIB=openssl --with-extralibs="-L/usr/lib64 -L/usr/lib64/mysql" --with-ffmpeg=/usr/share/ffmpeg CXXFLAGS="-D__STDC_CONSTANT_MACROS"

**Comment by connortechnology** [comment]
This may be fixed.  Please try current master

**Comment by sbmechanics** [comment]
I was able to get this fixed with the current master.
Steps:
1) I had things custom built, so I decided to start over.  I created a CentOS 6.6 VM with minimal configuration.  On first boot, I disabled selinux and iptables, and did a yum update.
2) To try and see if 1.28.1 would fix the issue, and to try to speed things up, I installed zmrepo-6-4.el6.noarch.rpm and did a yum zonemaster install.  After restoring my previous config in the web gui, the problems were still there.
3) I installed the EPEL and RPMFusion repos.
4) I did a blanket yum install per the readme:
yum install cmake bzip2-devel ffmpeg ffmpeg-devel gnutls-devel httpd libjpeg-turbo libjpeg-turbo-devel mysql-devel mysql-server pcre-devel \
perl-Archive-Tar perl-Archive-Zip perl-Convert-BinHex perl-Date-Manip perl-DBD-MySQL perl-DBI perl-Device-SerialPort perl-Email-Date-Format perl-IO-stringy \
perl-IO-Zlib perl-MailTools perl-MIME-Lite perl-MIME-tools perl-MIME-Types perl-Module-Load perl-Package-Constants perl-Sys-Mmap perl-Time-HiRes \
perl-TimeDate perl-YAML-Syck perl-X10 perl-URI-Encode php php-cli php-mysql x264 vlc-devel vlc-core \
libcurl libcurl-devel polkit-devel git
5) I linked the ffmpeg directories as per https://github.com/ZoneMinder/ZoneMinder/issues/304.
6) I did a git, then ./cmakecacheimport.sh /etc/zm.conf,  cmake -C zm_conf.cmake, and make.
7) I stopped zoneminder, did a make install, a zmupdate.pl -f, and started zoneminder.
8) On configuring the login options, everything appears working.

Thanks very much.
